spec/lib/administrate/page/collection_spec.rb

```diff
@@ -0,0 +1,286 @@
+          search_with_scope = Administrate::Search.new(resolver, "scope:new")
+
+          page = Administrate::Page::Collection.new(dashboard_with_scopes_hash,
+                                                    search: search_with_scope)
+          expect(page.scoped_groups).to eq([])
+        ensure
+          remove_constants :User
+        end
+      end
+    end
+  end
+
+  describe "#current_scope_of(group)" do
+    let(:search) { Administrate::Search.new(nil, "searched words") }
+
+    describe "with no scopes defined" do
+      it "returns nil" do
+        page = Administrate::Page::Collection.new(dashboard_without_scopes,
+                                                  search: search)
+        expect(page.current_scope_of(:scopes)).to eq(nil)
+      end
+    end
+
+    describe "with an Array of scopes" do
+      # scopes will be grouped with the :scopes key in Search#collection_scopes
+      it "returns the scope if is into the specified group (:scopes)" do
+        begin
+          class User
+            def self.old; end
+          end
+          resolver = double(resource_class: User,
+                            dashboard_class: DashboardWithAnArrayOfScopes)
+          search_with_scope = Administrate::Search.new(resolver, "scope:old")
+
+          page = Administrate::Page::Collection.new(dashboard_with_scopes_array,
+                                                    search: search_with_scope)
+          expect(page.current_scope_of(:scopes)).to eq('old')
+        ensure
+          remove_constants :User
+        end
+      end
+
+      it "returns nil if the scope isn't into the specified group (:scopes)" do
+        begin
+          class User
+            def self.old; end
+          end
+          resolver = double(resource_class: User,
+                            dashboard_class: DashboardWithAnArrayOfScopes)
+          search_with_scope = Administrate::Search.new(resolver, "scope:new")
+
+          page = Administrate::Page::Collection.new(dashboard_with_scopes_array,
+                                                    search: search_with_scope)
+          expect(page.current_scope_of(:scopes)).to eq(nil)
+        ensure
+          remove_constants :User
+        end
+      end
+    end
+
+    describe "with a Hash of scopes" do
+      it "returns the scope if is into the specified group" do
+        begin
+          class User
+            def self.old; end
+          end
+          resolver = double(resource_class: User,
+                            dashboard_class: DashboardWithAnArrayOfScopes)
+          search_with_scope = Administrate::Search.new(resolver, "scope:old")
+
+          page = Administrate::Page::Collection.new(dashboard_with_scopes_hash,
+                                                    search: search_with_scope)
+          # the :old scope is defined in the :other group (see spec_helper.rb).
+          expect(page.current_scope_of(:other)).to eq('old')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.